pkg/extensions/sync/destination.go

```diff
@@ -248,6 +248,8 @@ func (registry *DestinationRegistry) copyManifest(repo string, manifestContent [
 func (registry *DestinationRegistry) copyBlob(repo string, blobDigest digest.Digest, blobMediaType string,
 	tempImageStore storageTypes.ImageStore,
 ) error {
+	defer os.Remove(tempImageStore.BlobPath(repo, blobDigest))
```

Comment:
This seems to be a side effect of copyBlob which doesn't have anything to do with the actual copy, which may be confusing for developers.

I suggest either a :
- name change
- move the deletion outside the method and into a new method which would be called instead of this one.